src/mm/mod.rs

```diff
@@ -107,3 +103,19 @@ impl<'alloc, T: arch::ArchitectureMemory> MemoryManager<'alloc, T> {
         self.arch.reload();
     }
 }
+
+pub trait MemoryManager {
+    fn map(&mut self, to: usize, from: usize, perms: Permissions);
+    fn reload_pt(&mut self);
+
+}
+
+impl<T: arch::ArchitectureMemory> MemoryManager for MemoryManagement<'_, T> {
+    fn map(&mut self, to: usize, from: usize, perms: Permissions) {
+        self.arch.map(&mut self.page_manager, to, from, perms)
+    }
+
+    fn reload_pt(&mut self) {
```

Comment:
```suggestion
    fn reload_page_table(&mut self) {
```

Reply:
done in 9202e0e178b3dbfb6ea8129e22f142fd1896b98f